tests/posix/common/src/clock.c

```diff
@@ -46,34 +46,16 @@ void test_posix_clock(void)
 
 void test_posix_realtime(void)
 {
-	/* Make sure the realtime and monotonic clocks start out the
-	 * same.  This is not true on posix and where there is a
-	 * realtime clock, so don't keep this code.
-	 */
-
 	int ret;
 	struct timespec rts, mts;
 	struct timeval tv;
 
-	printk("POSIX clock set APIs\n");
-
-	/* Minimal sleep to align us to the next tick interval. This
-	 * helps with a case that 2 consecutive calls to clock_gettime()
-	 * below return different values. Note that the tick alignment
-	 * approach may break, in which case follow the suggestion in the
-	 * comment above.
-	 */
-	k_usleep(1);
-
 	ret = clock_gettime(CLOCK_MONOTONIC, &mts);
 	zassert_equal(ret, 0, "Fail to get monotonic clock");
 
 	ret = clock_gettime(CLOCK_REALTIME, &rts);
 	zassert_equal(ret, 0, "Fail to get realtime clock");
 
-	zassert_equal(rts.tv_sec, mts.tv_sec, "Seconds not equal");
```

Comment:
I'm not sure I would remove this one because if the seconds doesn't match then there is likely a problem that needs to be addressed or highlighted. The primary problem on this test case is the nanoseconds where the TICK counter has a good chance of incrementing between the two calls to clock_gettime() so the assumption that the calls would be equal is faulty. Having the test consider this would be useful but maybe we just drop that equal test or do what I suggested in the bug report and do a diff test.

Reply:
POSIX has no requirement that the seconds counter be equal between CLOCK_MONOTONIC and CLOCK_REALTIME and they most likely will be completely different.

It just happens to the the case in the implementation that they both start at zero.

So it's testing out an implementation detail rather than expected behaviour.

It's a poorly conditioned test but also is something that should not be relied on - e.g. if CLOCK_REALTIME were ever connected to an RTC the assumption would be immediately broken.